Fabric.Identity.API/scripts/Install-Identity.ps1

```diff
@@ -150,6 +136,64 @@ if ($fabricInstallerSecret){
+$idpssServiceUrl = Get-ApplicationEndpoint -appName $idpssConfigStore.appName -applicationEndpoint $idpssConfigStore.applicationEndPoint -installConfigPath $installConfigPath -scope $idpssSettingsScope -quiet $quiet
+    
+$idpssWebDeployParameters = Get-WebDeployParameters -serviceConfig $idpssConfigStore -commonConfig $commonConfigStore
+$idpssStandalonePath = ".\Fabric.IdentityProviderSearchService.zip"
+$idpssInstallerPath = "..\WebDeployPackages\Fabric.IdentityProviderSearchService.zip"
+$idpssInstallPackagePath = Get-WebDeployPackagePath -standalonePath $idpssStandalonePath -installerPath $idpssInstallerPath
+
+$idpssIISUser = Get-IISAppPoolUser -credential $credential -appName $idpssConfigStore.appName -storedIisUser $idpssConfigStore.iisUser -installConfigPath $installConfigPath -scope $idpssSettingsScope
+
+$idpssInstallApplication = Publish-DosWebApplication -WebAppPackagePath $idpssInstallPackagePath `
+                      -WebDeployParameters $idpssWebDeployParameters `
+                      -AppPoolName $idpssConfigStore.appPoolName `
+                      -AppPoolCredential $idpssIISUser.credential `
+                      -AuthenticationType "Windows" `
+                      -WebDeploy
+
+$idpssAppPoolUser = $idpssIISUser.UserName
+
+# uses Discovery url, we should be able to remove since I Get-IISAppPoolUser above
+$idpssName = "IdentityProviderSearchService"
+#$idpssAppPoolUser = Find-IISAppPoolUser -applicationName $idpssName -discoveryServiceUrl $discoveryServiceUrl -noDiscoveryService $true -quiet $quiet
+
+if($useAzure) {
+    Add-PermissionToPrivateKey $idpssAppPoolUser $selectedCerts.SigningCertificate read
+}
+
+$idpssDirectory = [io.path]::combine([System.Environment]::ExpandEnvironmentVariables($selectedSite.physicalPath), $idpssName)
+New-AppRoot $idpssDirectory $idpssAppPoolUser
+
+Add-DatabaseSecurity $idpssAppPoolUser $installSettings.identityDatabaseRole $identityDatabase.DbConnectionString
+
+Register-ServiceWithDiscovery -iisUserName $idpssAppPoolUser -metadataConnStr $metadataDatabase.DbConnectionString -version $idpssInstallApplication.version -serverUrl $idpssServiceUrl `
+-serviceName $idpssName -friendlyName "Fabric.IdentityProviderSearchService" -description "The Fabric.IdentityProviderSearchService searches Identity Providers for matching users and groups.";
+
+#$registrationApiSecret = Add-RegistrationApiRegistration -identityServerUrl $identityServiceUrl -accessToken $accessToken
+
+# uses discovery url similar to above
+#$idpssConfig = Get-WebConfigPath -applicationName $idpssName -discoveryServiceUrl $discoveryServiceUrl -noDiscoveryService $true -quiet $quiet
+$siteName = $idpssConfigStore.siteName
+$sitePath = $(get-item "iis:\sites\$siteName").physicalPath
+$idpssDirectory = [io.path]::combine([System.Environment]::ExpandEnvironmentVariables($sitePath), $idpssName)
+$idpssConfig = $idpssDirectory + "\web.config"
+
+Set-IdentityProviderSearchServiceWebConfigSettings -webConfigPath $idpssConfig `
```

Comment:
Do we need this? For web deploy based powershell installs, we use the -webDeployParameters parameter of Publish-DosWebApplication which performs the web.config transform for us.

Reply:
So do you mean we don't need to insert the azure app settings into the web.config. I believe that is part of our Identity and IdPSS azure registration script.